### PR TITLE
CXF-9082: Allow to modify sensitive header list

### PR DESCRIPTION
**Pull request by dmytro-sylaiev**
* Fix any sensitive information is printed to the logs

https://issues.apache.org/jira/browse/CXF-9082

### CODE REVIEW

File: rt/transports/http/src/main/java/org/apache/cxf/transport/http/Headers.java [hunk 1]
```diff
@@ -580,4 +580,8 @@ public static String toHttpDate(Date date) {
     public static String toHttpLanguage(Locale locale) {
         return locale.toString().replace('_', '-');
     }
+
+    public static List<String> getSensitiveHeaderList() {
```
Comment (reta) [inline rt/transports/http/src/main/java/org/apache/cxf/transport/http/Headers.java:584]: Please convert `SENSITIVE_HEADERS` to the contextual property:

```
private static final String SENSITIVE_HEADERS = "org.apache.http.sensitive.headers";


List<String> getSensitiveHeaders() {
    return MessageUtils.getContextualStrings(message, SENSITIVE_HEADERS, Set.of("Authorization", "Proxy-Authorization");
}
```

**Review by reta — CHANGES_REQUESTED** [review]

File: rt/transports/http/src/main/java/org/apache/cxf/transport/http/Headers.java (hunk 1 above)
Comment (dmytro-sylaiev) [reply rt/transports/http/src/main/java/org/apache/cxf/transport/http/Headers.java:584]: Hello @reta 
Both `toString()` and `logProtocolHeaders` methods (who are the only users of `SENSITIVE_HEADERS` list) are static and don't use `message` to print headers.

Comment (reta) [reply rt/transports/http/src/main/java/org/apache/cxf/transport/http/Headers.java:584]: Hello @dmytro-sylaiev , that is right, those are internal APIs (package protected) and has to be changed to instance methods (or accept message as an argument), thanks.

**Comment by dmytro-sylaiev** [comment]
Hello @reta 
I've changed the PR to make sensitive headers a contextual property as you asked, but now for me is not clear how to pass the value when use `WebClient:invoke`, seems the end-user can not modify the result Message properties

**Comment by reta** [comment]
> I've changed the PR to make sensitive headers a contextual property as you asked, but now for me is not clear how to pass the value when use `WebClient:invoke`, seems the end-user can not modify the result Message properties

Thanks @dmytro-sylaiev , there a multiple ways to provide these properties, to list just a few:
 - using `Bus` instance
 - using `WebClient.getConfig(wc).getRequestContext().put(...)` context

**Review by reta — APPROVED** [review]

**Pull request merged**